include/pybind11/detail/smart_holder_poc.h

```diff
@@ -138,15 +138,31 @@ inline bool is_std_default_delete(const std::type_info &rtti_deleter) {
            || rtti_deleter == typeid(std::default_delete<T const>);
 }
 
+// Meant to help detecting invalid `reinterpret_cast`s or similar.
+#ifdef PYBIND11_SMART_HOLDER_PADDING_ON
+#    define PYBIND11_SMART_HOLDER_PADDING(N) int PADDING_##N##_[11]
+#else
+#    define PYBIND11_SMART_HOLDER_PADDING(N)
+#endif
+
 struct smart_holder {
+    PYBIND11_SMART_HOLDER_PADDING(1);
     const std::type_info *rtti_uqp_del = nullptr;
+    PYBIND11_SMART_HOLDER_PADDING(2);
     std::shared_ptr<void> vptr;
+    PYBIND11_SMART_HOLDER_PADDING(3);
```

Comment:
padding between bool bitfields? Why use bitfields in any case?

Reply:
The padding was for brute-force stress-testing only while working on #5213. I just reverted it here (c02d2cd4ec0fdef028be5b11421ddc61157bef7b). It's easy enough to redo if/as needed.

(The bitfields are just to keep the memory footprint small.)